### PR TITLE
[google_sign_in_platform_interface] Add support for `serverClientId`

### PR DESCRIPTION
**Pull request by blaugold**
This PR is a prerequisite for implementing https://github.com/flutter/plugins/pull/5250.
It adds support for passing a server client ID to platform implementations when initializing them.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by stuartmorgan** [comment]
The main PR hasn't been reviewed yet; as [explained in the docs](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changing-federated-plugins), that's a pre-requisite to landing component PRs.

**Comment by stuartmorgan** [comment]
Marking as a draft for now to get this off of our review queue; once the main PR is approved please mark this one as ready for review.

**Comment by blaugold** [comment]
@stuartmorgan This PR is ready to review, in case you missed it.

**Review by stuartmorgan — APPROVED** [review]
LGTM!

@cyanglaz could you do the secondary review here?

File: packages/google_sign_in/google_sign_in_platform_interface/lib/src/types.dart [hunk 1]
```diff
@@ -49,9 +50,21 @@ class SignInInitParameters {
   /// By default, the list of accounts will not be restricted.
   final String? hostedDomain;
 
-  /// The client ID to use when signing in.
+  /// The OAuth client ID of the app.
+  ///
+  /// The default is null, which means that the client ID will be sourced from a
+  /// configuration file, if required on the current platform. A value specified
+  /// here takes precedence over a value specified in a configuration file.
```
Comment (cyanglaz) [inline packages/google_sign_in/google_sign_in_platform_interface/lib/src/types.dart:57]: nits: Could you add a link to how to set the values with configuration file?  Something like:
```
/// See also: 
/// 
/// [Platform Integration](https://github.com/flutter/plugins/tree/main/packages/google_sign_in/google_sign_in#platform-integration), where you can find the details about the configuration files.
```

**Review by cyanglaz — APPROVED** [review]
LGTM % nits

**Comment by stuartmorgan** [comment]
Hm, submit-queue is stuck on the wrong status for some reason; I'll land it manually.

**Pull request merged**